kernel/core/datamanagement/memory.cpp

```diff
@@ -3820,6 +3820,226 @@ std::vector<AnovaResult> Memory::getAnova(const VectorIndex& colCategories, size
+        centroids.push_back(new_value);
+
+        for(size_t i = 1; i < nClusters; i++)
+        {
+            // todo this loop could propably be also be included into the standard Algorithm Loop
+            // however this could be less readable because at more variables have to be added
+
+            // K++ Step 2: calculate distance to nearest centroid
+            std::vector<double> distance_vec;
+            for(size_t i = 0; i < col_size; i++)
+            {
+                std::vector<mu::value_type> value = readMem(VectorIndex({i}), columns);
+                double dis = calculateL2Norm(value, centroids[0]);
+                size_t idx = 0;
+                for(size_t j = 1; j < centroids.size(); j++)
+                {
+                    double dis2 = calculateL2Norm(value, centroids[j]);
+                    if(dis2 < dis){
+                        dis = dis2;
+                        idx = j;
+                    }
+                }
+                distance_vec.push_back(dis);
+                if(clusters[i] != mu::value_type(idx))
+                {
+                    clusters[i] = idx;
+                }
+            }
+
+            // K++ Step 3: add new centroid at point with highest distance to nearest neighbour
+            size_t max_elem_idx = std::distance(distance_vec.begin(), std::max_element(distance_vec.begin(), distance_vec.end()));
+            std::vector<mu::value_type> new_value = readMem(VectorIndex({max_elem_idx}), columns);
+            centroids.push_back(new_value);
+        }
+    }
+    else
+    {
+        // Run standard Init: Random Seeds
+        for(size_t i = 0; i < nClusters; )
+        {
+            size_t randinit = rand()%col_size;
+            std::vector<mu::value_type> new_value = readMem(VectorIndex({randinit}), columns);
+            if(isUnique(new_value, centroids))
+            {
+                centroids.push_back(new_value);
+                i++;
+            }
+        }
+    }
+
+    long double inertia = 0;
+    for(size_t iteration = 0; iteration < maxIterations; iteration++)
+    {
+        // Step 2 assign points to closest Cluster centroid
+        size_t change = 0;
+        inertia = 0;
+        for(size_t i = 0; i < col_size; i++)
+        {
+            std::vector<mu::value_type> value = readMem(VectorIndex({i}), columns);
+            double min_distance = calculateL2Norm(value, centroids[0]);
+            size_t idx = 0;
+            for(size_t j = 1; j < nClusters; j++)
+            {
+                double curr_distance = calculateL2Norm(value, centroids[j]);
+                if(curr_distance < min_distance){
+                    min_distance = curr_distance;
+                    idx = j;
+                }
+            }
+
+            idx++;
+            if(clusters[i] != mu::value_type(idx))
+            {
+                clusters[i] = idx;
+                change++;
+            }
+
+            inertia += std::pow(min_distance,2);

```

Comment:
```suggestion
            inertia += intPower(min_distance,2);
```
Recommended, because uses a `for` loop internally and not `std::log()`